depthai_ros_driver/test/conversion.cpp

```diff
@@ -0,0 +1,47 @@
+#include <depthai_ros_driver/conversion.hpp>
+
+#include <gtest/gtest.h>
+
+namespace rr {
+bool cvMatAreEqual(const cv::Mat& a, const cv::Mat& b) {
+    if (a.size() != b.size()) {
+        return false;
+    }
+
+    cv::Mat temp;
+    cv::bitwise_xor(a, b, temp);
+    return cv::countNonZero(temp.reshape(1)) == 0;
+}
+
+TEST(chw2hwc, base_comparison) {
+    // test that cvMatAreEqual is working as expected
+    uchar* a = new uchar[3]{1, 2, 3};
+    uchar* b = new uchar[3]{1, 2, 3};
+    uchar* c = new uchar[3]{4, 5, 6};
+
+    cv::Mat A(1, 1, CV_8UC1, a);
+    cv::Mat B(1, 1, CV_8UC1, b);
+    cv::Mat C(1, 1, CV_8UC1, c);
+
+    EXPECT_TRUE(cvMatAreEqual(A, B));
+    EXPECT_FALSE(cvMatAreEqual(B, C));
+    EXPECT_FALSE(cvMatAreEqual(A, C));
+}
+
+TEST(chw2hwc, tripleTrip) {
+    cv::Mat mat(3, 3, CV_8UC3);
+    uchar* mat_ptr = mat.data;
+    for (int i = 1; i <= 27; ++i) {
+        mat_ptr[i] = i;
+    }
+
+    cv::Mat once = chw2hwc(mat);
+    cv::Mat twice = chw2hwc(chw2hwc(mat));
+    cv::Mat thrice = chw2hwc(chw2hwc(chw2hwc(mat)));
+
+    EXPECT_FALSE(cvMatAreEqual(mat, once));
+    EXPECT_FALSE(cvMatAreEqual(mat, twice));
+    EXPECT_TRUE(cvMatAreEqual(mat, thrice));
```

Comment:
Is this true in all the image sizes? 
applying `chw2hwc` multiple times does not have meaning. It would make more sense to apply its inverse function and compare if the images are coming back.

Reply:
doing this 3 times should return it to original? Basically it's reordering the left dimension to the right dimension
chw -> hwc -> wch -> chw

Reply:
image size needs to be 3 x 3 x 3 for satisfying chw -> hwc -> wch -> chw. 
what is expected by `chw2hwc` is interleaving the pixels, and I don't know what `chw2hwc` means for already interleaved images.

Reply:
@ardiya can we check if this test passes for a 5x5x3 image?

Reply:
Oh, yeah it only works with 3x3x3 images.
Doing this 3 times on 5x5x3 doesn't satisfy chw->hwc->wch->chw

Reply:
> what is expected by chw2hwc is interleaving the pixels, and I don't know what chw2hwc means for already interleaved images.

Should I rename it to [interleaved2planar](https://github.com/luxonis/depthai-ros/blob/noetic-devel/depthai_bridge/src/ImageConverter.cpp#L182-L205) and [planar2interleaved](https://github.com/luxonis/depthai-ros/blob/noetic-devel/depthai_bridge/src/ImageConverter.cpp#L182-L205) like what luxonis did?

Reply:
> > what is expected by chw2hwc is interleaving the pixels, and I don't know what chw2hwc means for already interleaved images.
> 
> Should I rename it to [interleaved2planar](https://github.com/luxonis/depthai-ros/blob/noetic-devel/depthai_bridge/src/ImageConverter.cpp#L182-L205) and [planar2interleaved](https://github.com/luxonis/depthai-ros/blob/noetic-devel/depthai_bridge/src/ImageConverter.cpp#L182-L205) like what luxonis did?

`chw2hwc` also explains what the function does, but yes, if luxonis code has that function, it can be easier to grep the code  by the using same name.

Reply:
renamed, and fix test to interleaved->planar->interleaved and planar->interleaved->planar